src/app/screens/ConfirmPayment/index.tsx

```diff
@@ -5,44 +5,59 @@ import PaymentSummary from "@components/PaymentSummary";
 import PublisherCard from "@components/PublisherCard";
 import SuccessMessage from "@components/SuccessMessage";
 import lightningPayReq from "bolt11";
-import { useEffect, useRef, useState } from "react";
-import { useNavigate, useSearchParams } from "react-router-dom";
+import type { PaymentRequestObject, TagsObject } from "bolt11";
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
 import { toast } from "react-toastify";
 import { useAccount } from "~/app/context/AccountContext";
 import { useSettings } from "~/app/context/SettingsContext";
+import { useNavigationState } from "~/app/hooks/useNavigationState";
 import { USER_REJECTED_ERROR } from "~/common/constants";
 import msg from "~/common/lib/msg";
 import utils from "~/common/lib/utils";
 import { getFiatValue } from "~/common/utils/currencyConvert";
 import getOriginData from "~/extension/content-script/originData";
 import type { OriginData } from "~/types";
 
-export type Props = {
-  origin?: OriginData;
-  paymentRequest?: string;
-};
+type Invoice = PaymentRequestObject & { tagsObject: TagsObject };
 
-function ConfirmPayment(props: Props) {
-  const { isLoading: isLoadingSettings, settings } = useSettings();
-  const showFiat = !isLoadingSettings && settings.showFiat;
-
-  const [searchParams] = useSearchParams();
+function ConfirmPayment() {
+  const navState = useNavigationState();
   const navigate = useNavigate();
   const auth = useAccount();
 
-  const invoiceRef = useRef(
-    lightningPayReq.decode(
-      props.paymentRequest || (searchParams.get("paymentRequest") as string)
-    )
-  );
-  const originRef = useRef(props.origin || getOriginData());
-  const paymentRequestRef = useRef(
-    props.paymentRequest || searchParams.get("paymentRequest")
-  );
-  const [budget, setBudget] = useState(
-    ((invoiceRef.current?.satoshis || 0) * 10).toString()
-  );
+  // TODO: separate form from paymentRequest/invoice/origin handling
+
+  const [paymentRequest, setPaymentRequest] = useState("");
+  const [invoice, setInvoice] = useState<Invoice>();
+  const [origin, setOrigin] = useState<OriginData>();
+  const [budget, setBudget] = useState("");
   const [fiatAmount, setFiatAmount] = useState("");
+  const { isLoading: isLoadingSettings, settings } = useSettings();
+  const showFiat = !isLoadingSettings && settings.showFiat;
+
+  useEffect(() => {
+    if (navState.args?.paymentRequest && navState.origin) {
+      const paymentRequest = navState.args.paymentRequest as string;
+      setPaymentRequest(paymentRequest);
+      setInvoice(lightningPayReq.decode(paymentRequest));
+      setOrigin(navState.origin);
+    } else if (navState.paymentRequest) {
+      const paymentRequest = navState.paymentRequest;
+      setPaymentRequest(paymentRequest);
```

Comment:
This seems fine, but also including `setInvoice` or `setOrigin` here leads to this in the test:
```sh
  console.error
    Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.
        at ConfirmPayment (/src/app/screens/ConfirmPayment/index.tsx:29:59)
        at basename (/node_modules/react-router/umd/react-router.development.js:821:7)
        at basename (/node_modules/react-router/umd/react-router.development.js:734:7)

      43 |     //   } else if (navState.paymentRequest) {
      44 |     const paymentRequest = navState.paymentRequest;
    > 45 |     setPaymentRequest(paymentRequest);
         |     ^
      46 |     const invoice = lightningPayReq.decode(paymentRequest);
      47 |     // setInvoice(invoice);
      48 |     setOrigin(getOriginData());
```

In the extension everything seems working fine.
@lisabaut any ideas?
I don't see that the `navState` is being changed via the `useEffect`. Maybe it's the mock in the test file?

Reply:
Testing form component instead of index:
- works
- same test as before (with minor adjustment)